src/coreclr/nativeaot/System.Private.CoreLib/src/System/Type.NativeAot.cs

```diff
@@ -30,15 +31,8 @@ internal static unsafe RuntimeType GetTypeFromMethodTable(MethodTable* pMT)
             // unifier's hash table.
             if (MethodTable.SupportsWritableData)
             {
-                ref UnsafeGCHandle handle = ref Unsafe.AsRef<UnsafeGCHandle>(pMT->WritableData);
-                if (handle.IsAllocated)
-                {
-                    return Unsafe.As<RuntimeType>(handle.Target);
-                }
-                else
-                {
-                    return GetTypeFromMethodTableSlow(pMT, ref handle);
-                }
+                ref RuntimeType? type = ref Unsafe.AsRef<RuntimeType?>(pMT->WritableData);
+                return type ?? GetTypeFromMethodTableSlow(pMT, ref type);
```

Comment:
```suggestion
                return Unsafe.AsRef<RuntimeType?>(pMT->WritableData) ?? GetTypeFromMethodTableSlow(pMT);
```
We should fetch the writetable data slow in the slow method to minimize size of the aggressively inlined code.